doc/build.md

```diff
@@ -8,7 +8,7 @@ The OGDF build configuration is generated using [CMake](http://www.cmake.org/).
 
  * CMake 3.8+
  * C++17 compliant compiler
-   * gcc 7+
+   * gcc 8+
    * clang 6+
```

Comment:
Do we still support clang 6? Afaik we don't test any version except for 15. And I'm also not sure which version of MSVC GitHub Actions is currently using

Reply:
Good point. I think it's ok to not CI-test all versions mentioned here if we have tested them at least once manually. People can then complain if it's not working for some reason.

However, my main concern was to enable std::filesystem and I should really check that it works with all the versions we mention here...
* gcc 7 has std::filesystem, gcc 9.1 does not require linking with -lc++fs anymore.
* clang 7 has std::filesystem, clang 9.0 does not require linking with -lc++fs anymore.
* [Visual Studio 2017, Version 15.7](https://learn.microsoft.com/en-us/cpp/standard-library/filesystem?view=msvc-170), has std::filesystem, I don't know of any linking requirements.